src/app/courses/components/courses-list/courses-list.component.html

```diff
@@ -1,5 +1,5 @@
-<ul>
-  <li *ngFor="let course of courses; trackBy: trackByFn">
+<ul >
+  <li *ngFor="let course of (courses | orderBy | filter: query); trackBy: trackByFn">
```

Comment:
Could you please use one of the pipes inside the component?
https://www.angularjswiki.com/angular/how-to-use-angular-pipes-in-components-and-services/